packages/agreement-process/test/testUpgradeAgreement.ts

```diff
@@ -0,0 +1,1273 @@
+        stamps: {},
+        createdAt: TEST_EXECUTION_DATE,
+        consumerDocuments: [
+          {
+            id: unsafeBrandId<AgreementDocumentId>(
+              publishedDescriptor.interface.id
+            ),
+            name: publishedDescriptor.interface.name,
+            prettyName: publishedDescriptor.interface.prettyName,
+            contentType: publishedDescriptor.interface.contentType,
+            path: publishedDescriptor.interface.path,
+            createdAt: publishedDescriptor.interface.uploadDate,
+          },
+        ],
+      };
+
+      await addOneAgreement(agreementToBeUpgraded, postgresDB, agreements);
+
+      const eservice: EService = getMockEService(
+        agreementToBeUpgraded.eserviceId,
+        generateId<TenantId>(),
+        [deprecatedDescriptor, publishedDescriptor]
+      );
+      await addOneEService(eservice, eservices);
+
+      const newAgreementId = unsafeBrandId<AgreementId>(
+        await agreementService.upgradeAgreement(
+          agreementToBeUpgraded.id,
+          authData,
+          uuidv4()
+        )
+      );
+
+      expect(newAgreementId).toBeDefined();
+      const actualCreatedAgreementV1 = await getExpectedAgreementEventV1(
+        newAgreementId,
+        0,
+        {
+          type: "AgreementAdded",
+          stream_id: newAgreementId,
+          version: "0",
+        },
+        decodeAgreementV1FromEvent
+      );
+
+      expect(actualCreatedAgreementV1)
+        .property("createdAt")
+        .satisfy(
+          (createdAt: Date) =>
+            new Date(Number(createdAt)) &&
+            new Date(Number(createdAt)) <= TEST_EXECUTION_DATE
+        );
+
+      const expectedCreatedAgreementV1 = {
+        id: expect.any(String),
+        eserviceId: agreementToBeUpgraded.eserviceId,
+        descriptorId,
+        producerId: agreementToBeUpgraded.producerId,
+        consumerId: agreementToBeUpgraded.consumerId,
+        verifiedAttributes: agreementToBeUpgraded.verifiedAttributes,
+        certifiedAttributes: agreementToBeUpgraded.certifiedAttributes,
+        declaredAttributes: agreementToBeUpgraded.declaredAttributes,
+        consumerNotes: agreementToBeUpgraded.consumerNotes,
+        state: toAgreementStateV1(agreementState.draft),
+        createdAt: BigInt(TEST_EXECUTION_DATE.getTime()),
+        consumerDocuments: [],
+        stamps: {},
+      };
+
+      expect(actualCreatedAgreementV1).toMatchObject(
+        expectedCreatedAgreementV1
+      );
+
+      const actualAgreementDocumentAddedV1 = await getExpectedAgreementEventV1(
+        newAgreementId,
+        1,
+        {
+          type: "AgreementConsumerDocumentAdded",
+          stream_id: newAgreementId,
+          version: "1",
+        },
+        decodeAgreementDocumentV1FromEvent
+      );
+
+      expect(actualAgreementDocumentAddedV1).toMatchObject({});
+    });
+
+    it("should throw error tenant does not exist", async () => {
```

Comment:
Could you follow the same naming used for other tests (see other open PRs) where we specify the error name in the test string e.g., `"should throw an agreementNotFound error when the agreement does not exist"`

Reply:
✅  Done with [e0c85ce0](https://github.com/pagopa/interop-be-monorepo/pull/392/commits/e0c85ce0fe508d0d9d5ff3f2b224eed628eac975)